### PR TITLE
complete base pdf extractor

### PR DESCRIPTION
**Pull request by rvente**
* Extract Text from pdf, return as list of pages

### CODE REVIEW

**Comment by rvente** [comment]
View the [Script Output](https://gist.github.com/rvente/63da95fc22487982c2f1fd313b27a580)

360 is `"\n"` because the uploaded file is a scan, not an actual document. Some syllabi are far from perfect. The majority seem fine.

**Comment by boubascript** [comment]
> 360 is `"\n"` because the uploaded file is a scan

What do you mean by scan and not document?

**Review by boubascript — APPROVED** [review]
great start lgtm thanks!!

**Comment by rvente** [comment]
> > 360 is `"\n"` because the uploaded file is a scan
> 
> What do you mean by scan and not document?

It's a photoscan of the pdf. It contains no text data in the file.

**Comment by rvente** [comment]
> great start lgtm thanks!!

Thanks for the review. Merging now. NLP stuff will happen in another branch.

File: crawler/src/crawler/crawl_pdf.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+
+class DirCrawler(PdfCrawler):
+    def dir_to_index(self, dir_to_pdf_dir: Path) -> DefaultDict[str, List]:
+        """
+        Args:
+            dir_dir_pdf: like DIR in DIR/SUBDIR/EXAMPLE.pdf
+
+        Returns:
+            pd.DataFrame of ['fname', 'syllabus_text']
```
Comment (rvente) [inline crawler/src/crawler/crawl_pdf.py:63]: TODO: the type annotation here needs changing

**Comment by joshnatis** [comment]
> It's a photoscan of the pdf. It contains no text data in the file.

Are you counting on all of the syllabi being ["True PDFs"](https://pdf.abbyy.com/learning-center/pdf-types/)? Or are you gonna be doing OCR stuff on the ones that aren't

If not, what should we do with the syllabi that are image-only?

**Comment by joshnatis** [comment]
Also,

>Some syllabi are far from perfect.

Can you make a list of them and send it over? Maybe we can fix them with some online OCR tools like [Sandwich PDF](https://www.sandwichpdf.com/) (that's supposedly an alternative name for PDFs that are made searchable lmfao), or ask the professors for a proper copy.

**Comment by rvente** [comment]
The output of this script and its successor can be [found here](https://gist.github.com/rvente/63da95fc22487982c2f1fd313b27a580). Feel free to edit the original repo (the source of our content). I think on our part (e.g. on our site), we'll have a good way to mark missing or garbled syllabi, so certainly no urgency on that front. Something like 95+ percent accuracy is good enough for MVP status.

**Pull request merged**